[jit] Emit script function calls during tracing.

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#25089 [jit] Emit script function calls during tracing.**

Previously, when the tracer encountered a scripted function (or method), it
inlined the function into the graph. Now, we emit a CallFunction or
CallMethod node instead.

Differential Revision: [D16987936](https://our.internmc.facebook.com/intern/diff/D16987936)

suo committed:
[jit] Emit script function calls during tracing.

Previously, when the tracer encountered a scripted function (or method), it
inlined the function into the graph. Now, we emit a CallFunction or
CallMethod node instead.

suo committed:
Update on "[jit] Emit script function calls during tracing."


Previously, when the tracer encountered a scripted function (or method), it
inlined the function into the graph. Now, we emit a CallFunction or
CallMethod node instead.

Differential Revision: [D16987936](https://our.internmc.facebook.com/intern/diff/D16987936)

suo committed:
Update on "[jit] Emit script function calls during tracing."


Previously, when the tracer encountered a scripted function (or method), it
inlined the function into the graph. Now, we emit a CallFunction or
CallMethod node instead.

Differential Revision: [D16987936](https://our.internmc.facebook.com/intern/diff/D16987936)

suo committed:
Update on "[jit] Emit script function calls during tracing."


Previously, when the tracer encountered a scripted function (or method), it
inlined the function into the graph. Now, we emit a CallFunction or
CallMethod node instead.

Differential Revision: [D16987936](https://our.internmc.facebook.com/intern/diff/D16987936)

suo committed:
Update on "[jit] Emit script function calls during tracing."


Previously, when the tracer encountered a scripted function (or method), it
inlined the function into the graph. Now, we emit a CallFunction or
CallMethod node instead.

Differential Revision: [D16987936](https://our.internmc.facebook.com/intern/diff/D16987936)

zdevito commented on test/test_jit.py:
```diff
@@ -10547,6 +10546,7 @@ def traced_fn(x):
         self.assertTrue(len(list(traced_fn.graph.inputs())) == 1)
         FileCheck().check("aten::mm").check("aten::add").run(str(traced_fn.graph))
 
+    @_tmp_donotuse_dont_inline_everything
```
but it says donotuse

suo replied:
:'(

zdevito commented on torch/jit/__init__.py:
```diff
@@ -565,6 +565,8 @@ def graph_diagnostic_info():
                     break  # Graphs have already diverged
 
                 if n_mod.kind() == 'prim::Constant' and not (n_mod.mustBeNone() or n_check.mustBeNone()):
+                    if not n_mod.hasAttribute('value'):
```
why did this change?

zdevito reviewed (approved):
Looks good!

suo committed:
Update on "[jit] Emit script function calls during tracing."


Previously, when the tracer encountered a scripted function (or method), it
inlined the function into the graph. Now, we emit a CallFunction or
CallMethod node instead.

Differential Revision: [D16987936](https://our.internmc.facebook.com/intern/diff/D16987936)

suo committed:
Update on "[jit] Emit script function calls during tracing."


Previously, when the tracer encountered a scripted function (or method), it
inlined the function into the graph. Now, we emit a CallFunction or
CallMethod node instead.

Differential Revision: [D16987936](https://our.internmc.facebook.com/intern/diff/D16987936)

suo committed:
Update on "[jit] Emit script function calls during tracing."


Previously, when the tracer encountered a scripted function (or method), it
inlined the function into the graph. Now, we emit a CallFunction or
CallMethod node instead.

Differential Revision: [D16987936](https://our.internmc.facebook.com/intern/diff/D16987936)
